test/unit/StreamrClient.test.js

```diff
@@ -215,9 +218,12 @@ describe('StreamrClient', () => {
                     })
                     connection.expect(ResendLastRequest.create(sub.streamId, sub.streamPartition, sub.id, nbToResend))
                     connection.emitMessage(SubscribeResponse.create(sub.streamId))
+                    // sending second resend last request since no answer to the first request was received
+                    connection.expect(ResendLastRequest.create(sub.streamId, sub.streamPartition, sub.id, nbToResend))
+                    setTimeout(done, STORAGE_DELAY + 1000)
```

Comment:
In unit tests you should never use "real" time, but instead use [fake timers](https://sinonjs.org/releases/v1.17.6/fake-timers/) so that you have full control over time and can explicitly tick it forward (and the test will run fast).

Reply:
Same elsewhere in this file

Reply:
I managed to use fake timers in `Subscription.test.js` but in this file it was too complex. The `tick()` function must be called after the `setTimeout()` in `StreamrClient._resendAndSubscribe()` but there is no event emitted to to catch when that happens...